### ISSUE TITLE
IDE0072: False positive in switch statement with tuples

### ISSUE THREAD

**Issue by BieleckiLtd** [dotnet/roslyn-analyzers#7469]
### Analyzer

**Diagnostic ID**: [IDE0072](https://learn.microsoft.com/en-gb/dotnet/fundamentals/code-analysis/style-rules/ide0072): `Add missing cases to switch expression`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.303](https://dotnet.microsoft.com/download/dotnet/8.0)

### Describe the bug

![Image](https://github.com/user-attachments/assets/327027cf-50ec-4f00-b673-a4736d7954da)

The switch expression covers all possible cases, yet the analyzer requests a default case, which would be unreachable.

### Steps To Reproduce

```csharp
private bool isRed = true;
private bool isRound = false;
private string result => (isRed, isRound) switch
{
    (true, true) => "Red and round",
    (true, false) => "Red but not round",
    (false, true) => "Not red but round",
    (false, false) => "Not red and not round",
};
```

**Comment by Thomas-Shephard** [comment]
I believe this is issue should be under the [Roslyn repository](https://github.com/dotnet/roslyn/issues) as the area that causes this diagnostic is [PopulateSwitch](https://github.com/dotnet/roslyn/tree/main/src/Analyzers/Core/Analyzers/PopulateSwitch).

Although from a cursory look, I don't think this is particularly trivial issue to fix as the issue is because switches are currently only analyzed in a more detailed manner for Booleans and Enums. Tuples can contain other tuples and each tuple may contain up to 8 types.

See the [AnalyzeSwitch method](https://github.com/dotnet/roslyn/blob/main/src/Analyzers/Core/Analyzers/PopulateSwitch/AbstractPopulateSwitchDiagnosticAnalyzer.cs#L82)

**Comment by BieleckiLtd** [comment]
If I add default case, analyser can detect unreachable code and raises CS0162.
